Staking Wiki Doc - Reorg + writeup (#3623)
w3f/polkadot-wiki: docs/learn/learn-staking.md

keywords: [staking, stake, nominate, nominating, NPoS]
slug: ../learn-staking
---

import RPC from "./../../components/RPC-Connection"

Here you will lean about what is staking, why is important and how it works for
{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }}.

# Introduction

> Blockchain networks use [consensus](learn-consensus.md/#why-do-we-need-consensus) mechanisms to add

Reviewer: ```suggestion
Blockchain networks use [consensus](learn-consensus.md/#why-do-we-need-consensus) mechanisms to finalize
```

blocks on the chain. Consensus is the process of agreeing on something, in this case the progression

> of the blockchain or how blocks are added to the chain. Consensus is split into two protocols:

Reviewer: Better to show these two as list items

Author: Done

`block production`, i.e. the way multiple blocks candidates are produced, and `block finality`, i.e.

> the way only one block out of many candidates is selected and added to the chain. Because in public

Reviewer: You can always redirect the reader to another Wiki page that goes in depth on the topic https://wiki.polkadot.network/docs/learn-consensus#probabilistic-vs-provable-finality

```suggestion
the way only one block out of many candidates is selected and added to the canonical chain.
```

Author: Added link as requested

blockchains we have many participants who do not know each other (and probably never will), the
reach of consensus happens in a secure and trustless way using Proof-of-Work (PoW) or Proof-of-Stake

> (PoS). In PoS networks like {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} the

Reviewer: rephrasing

```suggestion
Proof-of-Work (PoW) and Proof-of-Stake(PoS) are well known mechanisms used to reach consensus in a secure and trustless way on public blockchains where we have many participants who do not know each other (and probably never will). In PoS networks like {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} the
```

security of the network is dependent on the amount of capital locked on chain: the more the capital

> locked the lower the chance that someone will be able to successfully attack the network, as they

Reviewer: ```suggestion
locked the lower the chance of an attack on the network, as the attacker
```

Here you will lean about what is staking, why is important and how it works for
{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }}.

## Introduction

Blockchain networks use [consensus](learn-consensus.md/#why-do-we-need-consensus) mechanisms to add
blocks on the chain. Consensus is the process of agreeing on something, in this case the progression
of the blockchain or how blocks are added to the chain. Consensus is split into two protocols:
`block production`, i.e. the way multiple blocks candidates are produced, and `block finality`, i.e.
the way only one block out of many candidates is selected and added to the chain. Because in public
blockchains we have many participants who do not know each other (and probably never will), the
reach of consensus happens in a secure and trustless way using Proof-of-Work (PoW) or Proof-of-Stake
(PoS). In PoS networks like {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} the
security of the network is dependent on the amount of capital locked on chain: the more the capital
locked the lower the chance that someone will be able to successfully attack the network, as they

> would require a lot of own tokens or to collude different network participants. The process of

Reviewer: Anyone who has significant stake in the network would not want to attack it, as it is not in their economic interests. Feel free to rephrase. The PoS network rewards good behaviour and punishes bad behaviour. Any rational actor staking on the network would want to get rewarded. It is not in their economic interests to orchestrate an attack on the network and risk losing their stake.

```suggestion
needs to incur heavy loss to orchestrate a successful attack. The process of
```

Author: Text added as requested

> locking tokens on the chain is also called `staking`. This is the main difference between PoS and

Reviewer: ```suggestion
locking tokens on the chain is called `staking`. 
```

> PoW networks like Bitcoin that base their security on solving mathematic puzzles, a solution that
> has been criticized due to the high amount of energy needed for computers to solve such puzzles. In
> PoW networks miners are responsible for adding blocks to the chain, and for doing such work they are
> rewarded with tokens. Similarly to the miners in PoW networks, in PoS networks we have `validators`.
> Token holders can lock funds on chain and for doing so they are getting `staking rewards`. There is

Reviewer: Feel free to rephrase

```suggestion
PoW networks miners are responsible for adding blocks to the chain, and for doing such work they are
rewarded with tokens. In order to do that, miners have to compete to solve difficult mathematic puzzles - a solution that has been criticized for the energy wastage. Similar to the 'miners' in PoW networks, PoS networks have `validators` who do not have to compete with each other to solve mathematical puzzles and are instead pre-selected to produce the blocks based on the stake backing them.
Token holders can lock funds on chain and for doing so they are getting `staking rewards`. There is
```

Author: Commit accepted and rephrased a bit

thus an economic incentive for token holders to become active participants who contribute to the
security and economic stability of the network. PoS networks in general are therefore more inclusive
than PoW networks, as participants do not need to have either the technical knowledge about
blockchain technology nor the experience in running mining equipment. PoS ensures that everybody has
"skin in the game" and thus can be held accountable. In case of misbehavior participants to the
staking process can be punished or `slashed`, and depending on the gravity of the situation their
stake can be partly or fully confiscated.

## Nominated Proof-of-stake overview

> The two main types of PoS are Delegated PoS (DPoS) and Nominated PoS (NPoS).
> 
> DPoS networks weigh validators by stake, oftentimes allowing the highest-stake validators to take
> disproportionate control of the network’s consensus protocol. In DPoS networks, delegators are not
> subject to loss of stake based on the behavior of the validator.

Reviewer: Probably better to avoid mention of DPoS right at the beginning. Instead, can highlight what NPoS is and how it differs from other PoS mechanisms like DPoS later on in the doc https://www.gemini.com/cryptopedia/proof-of-stake-delegated-proof-of-stake-consensus-mechanism

I'm leaning towards retaining existing Wiki introduction to NPoS

```suggestion
{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} implements NPoS (Nominated Proof-of-Stake),
a relatively novel and sophisticated mechanism to select the validators who are allowed to participate in 
its [consensus](learn-consensus.md) protocol. The NPoS encourages {{ polkadot: DOT :polkadot }}{{ kusama: KSM :kusama }} 
holders to participate as nominators.
```

keywords: [staking, stake, nominate, nominating, NPoS]
slug: ../learn-staking
---

import RPC from "./../../components/RPC-Connection"

> Here you will lean about what is staking, why is important and how it works for

Reviewer: ```suggestion
Here you will learn about what staking is, why it is important and how it works on
```

{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }}.

## Introduction

Blockchain networks use [consensus](learn-consensus.md/#why-do-we-need-consensus) mechanisms to finalize
blocks on the chain. Consensus is the process of agreeing on something, in this case the progression
of the blockchain or how blocks are added to the chain. Consensus consists into two actions:

- `block production`, i.e. the way multiple blocks candidates are produced, and 
- `block finality`, i.e. the way only one block out of many candidates is selected and added to the canonical chain (see [this](learn-consensus.md/#probabilistic-vs-provable-finality) article for more information about finality).

Proof-of-Work (PoW) and Proof-of-Stake (PoS) are well known mechanisms used to reach consensus in a secure and trustless way on public blockchains, where we have many participants who do not know each other (and probably never will). In PoW networks, miners are responsible for adding blocks to the chain, and for doing such work they are

> rewarded with tokens. Network security relies on the fact that to add blocks miners must compete to solve difficult mathematic puzzles - a solution that has been criticized for the energy wastage. In PoS networks like {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} the

Reviewer: ```suggestion
typically rewarded with tokens. Network security relies on the fact that the miners must compete to solve difficult mathematic puzzles to add blocks - a solution that has been criticized for the energy wastage. In PoS networks like {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} the
```

security of the network depends on the amount of capital locked on chain: the more the capital
locked the lower the chance of an attack on the network, as the attacker
needs to incur heavy loss to orchestrate a successful attack (more on this later on). The process of
locking tokens on chain is called `staking`. 

> Similar to the miners in PoW networks, PoS networks have `validators` who do not have to compete with each other to solve mathematical puzzles and are instead pre-selected to produce the blocks based on the stake backing them.

Reviewer: ```suggestion
Similar to the miners in PoW networks, PoS networks have `validators` but they do not have to compete with each other to solve mathematical puzzles and are instead pre-selected to produce the blocks based on the stake backing them.
```

> Token holders can lock funds on chain and for doing so they are getting `staking rewards`. There is

Reviewer: ```suggestion
Token holders can lock funds on chain and for doing so, they are getting `staking rewards`. There is
```

thus an economic incentive for token holders to become active participants who contribute to the
security and economic stability of the network. PoS networks in general are therefore more inclusive
than PoW networks, as participants do not need to have either the technical knowledge about
blockchain technology nor the experience in running mining equipment. 

> PoS ensures that everybody has

Reviewer: ```suggestion
PoS ensures that everybody participating in the staking process has
```

> "skin in the game" and thus can be held accountable. In case of misbehavior participants to the

Reviewer: ```suggestion
"skin in the game" and thus can be held accountable. In case of misbehavior participants in the
```

> staking process can be punished or `slashed`, and depending on the gravity of the situation their

Reviewer: ```suggestion
staking process can be punished or `slashed`, and depending on the gravity of the situation, their
```

> stake can be partly or fully confiscated. It is not in a staker's economic interest to orchestrate an attack and risk losing tokens. Any rational actor staking on the network would want to get rewarded, and the PoS network rewards good behavior and punishes bad behavior.

Reviewer: ```suggestion
stake can be partly or fully confiscated by the network. It is not in a staker's economic interest to orchestrate an attack and risk losing tokens. Any rational actor staking on the network would want to get rewarded, and the PoS network rewards good behavior and punishes bad behavior.
```

holders to participate as `nominators`. Nominators may back up to
{{ polkadot: <RPC network="polkadot" path="consts.staking.maxNominations" defaultValue={16}/> :polkadot }}
{{ kusama: <RPC network="kusama" path="consts.staking.maxNominations" defaultValue={24}/> :kusama }}
validators as trusted validator candidates, and the network will automatically distribute the stake among validators in
an even manner. Also, in NPoS the stake of both nominators and validators can be slashed.  For an in-depth review of NPoS check
[this](https://research.web3.foundation/en/latest/polkadot/NPoS/index.html) research article.

### Nominating validators

The action of nominating consists in a) locking or
bonding tokens (stake) on chain, and 2) nominate a set validator candidates to whom the stake will
be allocated. {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} uses tools ranging
from election theory to game theory to discrete optimization, to develop an efficient validator
selection process that offers fair representation and security, thus avoiding uneven power and

> influence among validators. The `election algorithm` used by {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} bases on the Proportional Justified Representation (PJR) method called [Phragmen](learn-phragmen.md). For more information about PJR methods visit [this](https://research.web3.foundation/en/latest/polkadot/NPoS/1.%20Overview.html?highlight=proportional%20justified%20representation#) research article.

Reviewer: Phragmen algorithm is one of the many algorithms that can be used in Polkadot NPoS. We probably have to change the URL for this Wiki doc https://wiki.polkadot.network/docs/learn-phragmen to learn-NPoS and avoid this ambiguity. For more context, check the discussion here https://github.com/w3f/polkadot-wiki/issues/3461

```suggestion
influence among validators. The election algorithms used by {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} are based on the Proportional Justified Representation (PJR) methods like [Phragmen](learn-phragmen.md). For more information about PJR methods visit [this](https://research.web3.foundation/en/latest/polkadot/NPoS/1.%20Overview.html?highlight=proportional%20justified%20representation#) research article.
```

Author: @DrW3RK should I open and issue about the URL?

Reviewer: Sure! We will have to add a redirect for it. It may have already been referenced on multiple third party websites

production during a specific `era`. An era is a period of time of {{ polkadot: 24 hours :polkadot }}{{ kusama: 6 hours :kusama }} during
which an `active set` of validators is producing blocks and performing other actions on chain. This
means that not all validators are in the active set, and such set changes between eras. Each era is
divided into 6 epochs or `sessions` during which validators are assigned as block producers to
specific time frames or `slots`. This means that validators know the slots when they will be
required to produce a block within a specific session, but they do not know all the slots within a
specific era. Having sessions adds a layer of security because it decreases the chance of having
multiple validators assigned to a slot colluding to harm the network.

### Rewards

Validators who produce a block are rewarded with tokens, and they can share rewards with their
nominators. Both validators and nominators can stake their tokens on chain and receive staking
rewards at the end of each era. The staking system pays out rewards equally to all validators
regardless of stake. Thus, having more stake on a validator does not influence the amount of block rewards

> it receives. This avoids centralization of power to few validators. There is a probabilistic

Reviewer: ```suggestion
it receives. This avoids centralization of power to a few validators. There is a probabilistic
```

component in the calculation of rewards, so they may not be exactly equal for all validators. In
fact, during each era validators can earn `era points` by doing different tasks on chain. The more
the points, the higher the reward for a specific era. This promotes validators' activity on chain.
To know more about era points, how and on which basis they are distributed visit the
[dedicated page](../maintain/maintain-guides-validator-payout.md). Distribution of the rewards are
pro-rata to all stakers after the validator's commission is deducted.

> ### Skin in the game

Reviewer: Search Engine Friendly subsections. Show up when searching about "staking"

```suggestion
### Skin in the game when Staking
```

The security of PoS networks depends on the amount of staked tokens. This means
that to successfully attack the network one would need a large amount of tokens that can be accrued

> by one single participant alone or by colluding different participants to act maliciously. In case

Reviewer: ```suggestion
by one single participant alone or would need different participants to collude and act maliciously. In case
```

of NPoS, if there is an attack both the validator(s) and nominators will be

> `slashed` and in some cases their stake partially or fully confiscated. So there is little interest of acting in a harmful way because all participants can be

Reviewer: ```suggestion
`slashed` and in some cases their stake partially or fully confiscated by the network and deposited to the treasury. So there is little interest of acting in a harmful way because all participants can be
```

> held accountable for bad actions. Also, in NPoS validators are paid equal rewards regardless from

Reviewer: ```suggestion
held accountable for bad actions. In NPoS, validators are paid equal rewards regardless from
```

> the amount they have at stake, avoiding thus large payouts to few large validators which would

Reviewer: ```suggestion
the amount they have at stake, thus avoiding large payouts to few large validators which would
```

ultimately lead to consensus centralization.

## Being a nominator

### Tasks and responsibilities

Since validator slots are limited, most of those who wish to stake their DOT and contribute to the
economic security of the network will be nominators, thus here we focus on the role of nominators.
However, it is worth mentioning that validators do most of the heavy lifting: they run hardware

> equipment and manage

Reviewer: Could be on their own hardware or on the cloud

```suggestion
However, it is worth mentioning that validators do most of the heavy lifting: they run the validator nodes and manage
```

produce new block candidates in [BABE](learn-consensus.md/#block-production-babe), vote and come to
consensus in [GRANDPA](learn-consensus.md/#finality-gadget-grandpa), validate the state transition
function of parachains, and possibly some other responsibilities regarding data availability and
[XCM](learn-cross-consensus.md). For more information you can take a look at the
[validator docs](learn-validator.md) to understand what you need to do as a validator. If you want
to be come a validator check [this](../maintain/maintain-guides-how-to-validate-polkadot.md) guide.

Nominators have far fewer responsibilities than validators. Those include selecting validators and
monitoring their performance, keeping an eye on changing commission rates (a validator can change
commission at any time), and general health monitoring of their validators' account. Thus, while not
being completely set-it-and-forget-it, a nominator's experience is relatively hands-off compared to
a validators. For more information you can take a look at the [nominator guide](learn-nominator.md)
to understand your responsibilities as a nominator.

> If you want to be come a nominator check

Reviewer: ```suggestion
If you want to become a nominator, check
```

[this](../maintain/maintain-guides-how-to-nominate-polkadot.md) guide. If you are a beginner and

> would like to securely stake your tokens using Polkadot JS Apps, watch [this](https://support.polkadot.network/support/solutions/articles/65000168057-how-do-i-stake-nominate-on-polkadot-) support article. {{ kusama: The video demonstrates it on Polkadot, but the procedure is the same for Kusama :kusama }}

Reviewer: ```suggestion
would like to securely stake your tokens using Polkadot JS Apps, refer to [this](https://support.polkadot.network/support/solutions/articles/65000168057-how-do-i-stake-nominate-on-polkadot-) support article. {{ kusama: The tutorial presented in the support article is demonstrated on Polkadot, but the procedure is the same for Kusama :kusama }}
```

### Selection of validators

> Choosing validators is not a simple task and it should take into account for nominator reward and

Reviewer: ```suggestion
The task of choosing validators is not simple, as it should take into account for nominator reward and
```

risk preferences. Ideally one aims to maximize the reward-to-risk ratio by maximizing rewards and minimizing risks, with sometimes having to
compromise between the two as minimizing risks might decrease rewards as well. Nominators should pay attention especially to six criteria when nominating validators (not in order of importance):

- amount of era points
- validator pool, equivalent to own stake (i.e. coming from the validator) + other stake (i.e. coming from nominators)

> - own stake

Reviewer: Is this the own stake of the nominator? If so, please mention that explicitly

Author: @DrW3RK it is the own stake coming from the validator who nominates himself (specified in brackets). Should I re-phrase?

Reviewer: Yeah, you can rephrase it something along these lines

- Recent history of the era points earned across eras
- Validator's self stake (Shows skin in the game)
- Total stake backing the Validator  (which is the sum of self stake and the stake coming from nominators)

Reviewer: Avoid usage of terms like Validator pool (given that pools are now associated with Nomination pools)

Author: Changed it as suggested

Reviewer: If you made those changes locally, commit them to the repo

- commission fees (i.e. how much validators charge nominators)
- verified identity
- previous slashes

The diagram below shows how the selection of those criteria affects reward-to-risk ratio.

![rewards and risks diagram](../assets/staking/Reward-risk%20nominating.png)

#### In theory

> To maximize rewards and minimize risk one could select those validators

Reviewer: ```suggestion
To maximize rewards and minimize risk, one could select those validators
```

that:

- have era points above average (because they will get more rewards for being active), 

> - have validator pool below average (because they will pay out more rewards per staked {{ polkadot: DOT :polkadot }}{{ kusama: KSM :kusama }}), 

Reviewer: The word pool can be misleading, especially with the nomination pools being rolled out

```suggestion
- have the total stake backing the validator below the average active validator stake (because they will pay out more rewards per staked {{ polkadot: DOT :polkadot }}{{ kusama: KSM :kusama }}), 
```

- have high own stake (because if slashed they have a lot to lose), 

> - have low commission fees but not 0% (because it makes sense that for doing the heavy lifting validators ask for a small commission), 

Reviewer: ```suggestion
- have low commission fees but not 0% (because it makes sense that for doing the heavy lifting, validators ask for a small commission), 
```

> - have on-chain registered identity (because increases trust and you can reach out to them or look at their website or social media), 

Reviewer: ```suggestion
- have on-chain registered identity (because it adds a layer of trust and possibly provides access to their website and contact details), 
```

- and have not been slashed (meaning that their on-chain behavior is genuine).

#### In practice

:::caution Nominators must periodically check their validators

> Nominating is _not_ a "set and forget" operation. The whole NPoS system is dynamic and as such nominators should

Reviewer: ```suggestion
Nominating is _not_ a "set and forget" operation. The whole NPoS system is dynamic and nominators should
```

periodically monitor the performance and reputation of their validators. Failing to do so could

> result in applied slashes and/or rewards not being paid out, even for a prolonged period of time.

Reviewer: ```suggestion
result in applied slashes and/or rewards not being paid out, possibly for a prolonged period of time.
```

:::

> Although the theory can be used as a general guideline, the practice is more complicated and

Reviewer: ```suggestion
Although the theory can be used as a general guideline, in practice it is more complicated and
```

> following the theory might not necessarily lead to the desired result. Validators might have validator pool below average, low commission and above average era points in one era and then having a totally

Reviewer: ```suggestion
following the theory might not necessarily lead to the desired result. Validators might have the total stake backing them below average, low commission and above average era points in one era and then having a totally
```

> different profile in the next one. There are criteria that vary more than others with era points
> being the most variable and thus the probabilistic component of staking rewards. It is not recommended
> to change nominations at each era only because the era points of one validator were low. Selection
> based on identity, no slashes and low commission should be fine. Variability in rewards due to the
> other criteria should level out over time. There might be the case when one validator always gets
> below average era points, this is a case when selecting another validator might help to increase the
> stake performance.

Reviewer: Rephrasing this paragraph. If I modified the original intent/ideas presented, feel free to make updates.

```suggestion
different profile in the next one. Selection based the criteria like on-chain identity, slash history and low commission make the staking rewards deterministic. But there are criteria that vary more than others, with era points
being the most variable and thus one of the key probabilistic components of staking rewards. It is not recommended
to change nominations because of the low era points of a validator in a single era. Variability in rewards due to the era points should level out over time. If a validator consistently gets
era points below average, it makes sense to nominate a better performing validator for the health of the network and for increased staking rewards.
```

> Check
> [this](https://support.polkadot.network/support/solutions/articles/65000150130-how-do-i-know-which-validators-to-choose-)
> support article to understand in detail how to select your own set of validators.

Reviewer: ```suggestion
Check
[this](https://support.polkadot.network/support/solutions/articles/65000150130-how-do-i-know-which-validators-to-choose-)
support article to understand in detail how to select the set of validators to nominate.
```

### Accounts

There are two different accounts that can be used to securely manage your funds while staking.

- **Stash:** This account holds funds bonded for staking, but delegates some functions to a
  controller account. As a result, you may actively participate to staking with a stash private key kept in

>   a cold wallet like ledger, meaning it stays offline all the time. Stash account keys are used to
>   sign staking actions such as bonding and unbonding funds.

Reviewer: You can unbond funds using controller

```suggestion
  a cold wallet like ledger, meaning it stays offline all the time. Stash account keys are used to
  sign staking actions such as bonding additional funds.
```

- **Controller:** This account acts on behalf of the stash account, signalling decisions about

>   nominating and validating. It sets preferences like commission (for validators) and payout

Reviewer: ```suggestion
  nominating and validating. It can set preferences like commission (for validators) and the staking rewards payout
```

  account. The earned rewards can be bonded (locked) immediately for staking on your stash account,
  which would effectively compound the rewards you receive over time. You could also choose to have
  them deposited to your controller account or a different account as free (transferable) balance.

>   If you are a validator, it also sets your [session keys](learn-keys.md#session-keys). Controller

Reviewer: ```suggestion
  If you are a validator, it can also be used to set your [session keys](learn-keys.md#session-keys). Controller
```

>   accounts only needs enough funds to pay for transaction fees.

Reviewer: ```suggestion
  accounts only need sufficient funds to pay for the transaction fees.
```

:::warning

> Never leave high balance on a controller account, such accounts are usually "hot" meaning that the

Reviewer: ```suggestion
Never leave high balance on a controller account which are usually "hot" as their
```

> private key is stored on the device (PC, phone) and it is exposed to the internet all the time you
> are using it. It is good practice to deposit rewards on the stash account or to send them to another
> cold account.

Reviewer: ```suggestion
private key is stored on the device (PC, phone) and it is always exposed to the internet for potential hacks and scams. It is good practice to deposit rewards on the stash account or to send them to another
account on a cold wallet.
```

  sign staking actions such as bonding additional funds.

- **Controller:** This account acts on behalf of the stash account, signalling decisions about
  nominating and validating. It can set preferences like commission (for validators) and the staking rewards payout
  account. The earned rewards can be bonded (locked) immediately for staking on your stash account,
  which would effectively compound the rewards you receive over time. You could also choose to have
  them deposited to your controller account or a different account as free (transferable) balance.
  If you are a validator, it can also be used to set your [session keys](learn-keys.md#session-keys). Controller
  accounts only need sufficient funds to pay for the transaction fees.

:::warning

Never leave high balance on a controller account which are usually "hot" as their
private key is stored on the device (PC, phone) and it is always exposed to the internet for potential hacks and scams. It is good practice to deposit rewards on the stash account or to send them to another
account on a cold wallet.

:::

![staking](../assets/NPoS/staking-keys_stash_controller.png)

> This hierarchy of separate key types for stash and controller accounts was designed so that nominators and validator operators can
> protect themselves much better than in systems with only one key. As a rule, the more often one uses
> a private key the higher its visibility and thus the chance it can be stolen. So, if one uses a key

Reviewer: ```suggestion
This hierarchy of separate keys for stash and controller accounts was designed to add a layer of protection to nominators and validator operators. The more often one exposes and uses a private key, the higher its vulnerability for hacks or scams. So, if one uses a key
```

> for multiple roles on a blockchain network, security can be easily compromised as the likelihood one
> uses that key often is high. Note that the damage linked to stolen private keys is different

Reviewer: ```suggestion
for multiple roles on a blockchain network, it is likely that the account can get compromised. Note that the damage linked to stolen private keys is different
```

depending on the type of account derivation. In case of soft derivation all derived accounts are
compromised. More information about account derivation can be found
[here](../learn/learn-accounts.md/#derivation-paths).

:::info

For ledger users staking directly on Ledger Live, currently there is only one option to use one

> account as both stash and controller.

Reviewer: ```suggestion
For ledger users staking directly on Ledger Live, currently there is no option to use separate stash and controller accounts.
```

:::

### Staking proxies

{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} is built using

> [substrate](https://substrate.io/), a modular system to efficiently build blockchains without having necessarily deep knowledge about blockchain technology. Within each module or `pallet` one can `call` different

Reviewer: ```suggestion
[substrate](https://substrate.io/), a modular system to efficiently build blockchains. Within each module or `pallet` one can `call` different
```

functions that have similar logic. For example, the staking pallet contains all functionalities
related to staking such as bonding or unbonding funds. The combined information of pallets and calls
constitutes an `extrinsic`, i.e. a transaction that is executed from outside the chain but that

> triggers an event within the chain. Continuing with the staking example, within the staking pallet

Reviewer: ```suggestion
triggers an event on the chain. Continuing with the staking example, within the staking pallet
```

one can bond funds and nominate some validators. The signature of such
extrinsic might trigger a reward payout at the end of an era; this is an event inside the chain. This way of having

> transactions categorized within pallets and functionalities makes it possible to create accounts

Reviewer: Are you sure about bonding funds and nominating validators triggers rewards payouts? I think the idea you are trying to convey here will need rephrasing.

> having special permissions also called **proxy accounts**.

Reviewer: Add link to the Proxy Wiki doc

> In {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} proxy accounts are special
> accounts to which one can delegate signatures to calls from specific pallets. There is thus the

Reviewer: ```suggestion
In {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} proxy accounts are special
accounts which can sign extrinsic calls made to specific pallets on behalf of the proxied account. There is thus the
```

> possibility to create staking proxy accounts that can be used to sign only calls from staking,
> session and utility pallets. This makes the stash account even more isolated than using a controller

Reviewer: ```suggestion
possibility to create staking proxy accounts that can be used to sign only extrinsic calls to staking,
session and utility pallets. This makes the stash account even more isolated than using a controller
```

> account since now one can bond / unbond / bond more funds using the staking proxy account. However,

Reviewer: ```suggestion
account since one can bond / unbond / bond more funds using the staking proxy account. However,
```

it is important to remember that actions on proxy accounts are limited, and in the case of staking
proxy accounts calls from the balance pallet cannot be signed. This means that it is not possible to

> send funds from a staking proxy. To do that one needs to remove that account as a staking proxy.

Reviewer: ```suggestion
it is important to remember that actions that can be performed by the proxy accounts are limited, and in the case of staking
proxy, extrinsic calls to the balances pallet cannot be signed. This means it is not possible to do balance transfers
on the proxied account through a staking proxy.
```

:::

> The solution of the election of nominators has to meet certain requirements, such as maximizing the

Reviewer: "The solution of the election of nominators" is ambiguous. The election solution comprises of both nominators and validators. It is primarily for electing validators for the active set

```suggestion
The "election solution" which is a connected graph between nominators and validators with the stake as edge weights, has to meet certain requirements, such as maximizing the
```

amount of stake to nominate validators and distributing the stake backing validators as evenly as
possible. The objectives of this election mechanism are to maximize the security of the network, and
achieve fair representation of the nominators. If you want to know more about how NPoS works (e.g.
election, running time complexity, etc.), please read
[here](http://research.web3.foundation/en/latest/polkadot/NPoS.html).

### Rewards distribution

:::info

The general rule for rewards across validator pools is that two validator pools get paid essentially
the same amount of tokens for equal work, i.e. they are not paid proportional to the stakes in
each pool. There is a probabilistic component to staking rewards in the form of
[era points](../maintain/maintain-guides-validator-payout.md##era-points) and
[tips](learn-transaction-fees.md#fee-calculation) but these should average out over time.

:::

> Validator pools are paid the same regardless of stake level. Pools with less stake will generally pay more to nominators

Reviewer: ```suggestion
Validators are paid the same regardless of stake backing them. Validators with less stake will generally pay more to nominators
```

> per-token than pools with more stake. This gives nominators an economic incentive to gradually shift their preferences to lower staked

Reviewer: ```suggestion
per-token than the ones with more stake. This gives nominators an economic incentive to gradually shift their preferences to lower staked
```

> validators that gain a sufficient amount of reputation. A consequence of this is that the stake across validator pools will be as evenly distributed as possible, to avoid a concentration of power among a few validators. In the long term, validator pools will have similar levels of stake, with the stake

Reviewer: ```suggestion
validators that gain a sufficient amount of reputation. A consequence of this is that the stake across validators will be as evenly distributed as possible which avoids concentration of power among a few validators. In the long term, validators will have similar levels of stake, with the stake
```

> being higher for higher reputation validators (meaning that a nominator that is willing to risk more
> by backing a validator with a low reputation will get paid more).

Reviewer: ```suggestion
being higher for validators with higher reputation. A nominator who is willing to risk more
by backing a validator with a lower reputation will get paid more, provided there are no slashing events.
```

Before distributing rewards to nominators, validators can create a cut of the reward (a commission) that is not shared with the nominators.
This cut is a percentage of the block reward, not an absolute value. After the commission gets

> deducted, the remaining portion is based on their staked value and split between the validator and
> all of the nominators who have voted for this validator.

Reviewer: ```suggestion
deducted, the remaining portion is distributed pro-rata based on their staked value and split between the validator and
all of the nominators whose stake has backed this validator.
```

For example, assume the block reward for a validator is 10 DOT. A validator may specify
`validator_commission = 50%`, in which case the validator would receive 5 DOT. The remaining 5 DOT
would then be split between the validator and their nominators based on the proportion of stake each

> nominator had. Note that validators can put up their own stake, and for this calculation, their
> stake acts just as if they were another nominator.

Reviewer: ```suggestion
nominator had. Note that for this calculation, validator's self-stake acts just as if they were another nominator.
```

Within a validator pool, a percentage of the reward goes thus to pay the validator's
commission fees and the remainder is paid pro-rata (i.e. proportional to stake) to the
nominators and validator. If a validator's commission is set to 100%, no tokens will be paid out to any
nominations in the validator pool. Notice in particular that the validator is rewarded twice: once in
commission fees for validating (if their commission rate is above 0%), and once for nominating

> itself with stake.

Reviewer: ```suggestion
Thus, a percentage of the reward goes thus to pay the validator's
commission fees and the remainder is paid pro-rata (i.e. proportional to stake) to the
nominators and validator. If a validator's commission is set to 100%, no tokens will be paid out to any
nominations in the validator pool. Notice in particular that the validator is rewarded twice: once in
commission fees for validating (if their commission rate is above 0%), and once for nominating
itself with stake.
```

than those in pool A because pool A has more overall stake. Sam has staked 50 DOT in pool A, but he

> only gets 8.3 in return, whereas Kitty gets 12.5 with the same amount of stake.

Reviewer: If you can, try to delete the references to "pools" in the doc and replace them with an appropriate wording

other nominators are essentially "wasting" their stake - they used their nomination to elect that
validator to the active stake, but receive no rewards in exchange for doing so.

> Note that the network slashes a validator slot for a misbehavior (e.g. validator

Reviewer: ```suggestion
Note that the network slashes a validator for a misbehavior (e.g. validator
```

offline, equivocation, etc.) the slashed amount is a fixed percentage (and not a fixed amount),
which means that validator pools with more stake get slashed more DOT. Again, this is done to
provide nominators with an economic incentive to shift their preferences and back less popular
validators whom they consider to be trustworthy.

> Also note that each validator candidate is free to name their desired commission

Reviewer: ```suggestion
Also, note that each validator candidate is free to name their desired commission
```

that all validators will need to be cost efficient to remain competitive, and that validators with
higher reputation will be able to charge slightly higher commission fees (which is fair).

### Claiming rewards

{{ kusama: Note that Kusama runs approximately 4x as fast as Polkadot, except for block production times.
Polkadot will also produce blocks at approximately six second intervals. :kusama }}

Rewards are recorded per session (approximately {{ polkadot: 4 hours :polkadot }}{{ kusama: one hour :kusama }}) and
calculated per era (approximately {{ polkadot: 24 hours :polkadot }}{{ kusama: 6 hours :kusama }}). Thus,
rewards will be calculated {{ polkadot: once per day :polkadot }}{{ kusama: 4 times per day :kusama }}. Rewards are calculated based on era points, which have a probabilistic component. In other words,
there may be slight differences in your rewards from era to era, and even amongst validators in the
active set at the same time. These variations should cancel out over a long enough timeline. See the
page on [Validator Payout Guide](../maintain/maintain-guides-validator-payout.md).

> In order to be paid your staking rewards, someone must claim them for each validator that you
> nominate. Staking rewards are kept available for 84 eras, which is approximately {{ polkadot: 84 days :polkadot }}{{ kusama: 21 days :kusama }}. For more information on why this is so, see the page on [simple payouts](learn-simple-payouts.md).

Reviewer: > In order to be paid your staking rewards, someone must claim them for each validator that you
nominate.

A bit ambiguous

```suggestion
The distribution of staking rewards to the nominators is not automatic and needs to be triggered by someone. Typically the validators take care of this, but anyone can permissionlessly trigger rewards payout for all the nominators whose stake has backed a specific validator in the active set of that era. Staking rewards are kept available for 84 eras, which is approximately {{ polkadot: 84 days :polkadot }}{{ kusama: 21 days :kusama }}. For more information on why this is so, see the page on [simple payouts](learn-simple-payouts.md).
```

:::info Payouts

Payouts are unclaimed rewards waiting to be paid out to both validators and nominators. If you go to the Staking payouts page on
[Polkadot-JS](https://polkadot.js.org/apps/#/staking), you will see a list of all validators
that you have nominated in the past 84 eras and for which you have not yet received a payout. The payout page is visible only to stakers.

:::

Each validator as well as their nominators have the option to trigger the payout for all unclaimed eras. Note that this will pay everyone
who was nominating that validator during those eras. Therefore, you may not
see anything in this tab, yet still have received a payout if somebody (generally, but not
necessarily, another nominator or the validator operator) has triggered the payout for that
validator for that era.

> :::warning Time limit

Reviewer: ```suggestion
:::warning Time limit to claim staking rewards
```

> If nobody claims your staking rewards within 84 eras, then you will not be able to claim them and they will be lost. Additionally, if the validator unbonds all their own stake, any pending payouts will also be lost. Since unbonding takes {{ polkadot: 28 days :polkadot }}{{ kusama: 7 days :kusama }}, nominators should check if they have pending payouts at least this often.

Reviewer: ```suggestion
If nobody claims your staking rewards within 84 eras, you will not be able to claim them and they will be lost. Additionally, if the validator unbonds all of their self stake, any pending payouts will also be lost. Since unbonding takes {{ polkadot: 28 days :polkadot }}{{ kusama: 7 days :kusama }}, nominators should periodically check if they have pending payouts.
```

:::

> Rewards can be directed to the same account used to sign the payout (controller), to the stash account (and either

Reviewer: ```suggestion
Rewards can be directed to the same account used to sign the payout (controller) or to the stash account (and either
```

guidelines for different levels of severity for offences. To understand how slash amounts are
calculated, see the equations in the section below.

- Level 1: isolated unresponsiveness, i.e. being offline for an entire session. Generally no
  slashing, only [_chilling_](#chilling).
- Level 2: concurrent unresponsiveness or isolated equivocation, it slashes a very small amount of the
  stake and chills.
- Level 3: misconducts unlikely to be accidental, but which do not harm the network's security to
  any large extent. Examples include concurrent equivocation or isolated cases of unjustified voting
  in [GRANDPA](learn-consensus.md). Slashes a moderately small amount of the stake and chills.
- Level 4: misconduct that poses a serious security or monetary risk to the system, or mass
  collusion. Slashes all or most of the stake behind the validator and chills.

> For more information about slashing visit the dedicated page.

Reviewer: Link to the Wiki doc
